src/pages/SignUp/signUp.test.tsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import { render, waitFor, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import SignUpForm from 'components/SignUpForm';
+import { useCreateUserMutation } from 'services/blackMarketApi';
+
+jest.mock('services/blackMarketApi', () => ({
+  useCreateUserMutation: jest.fn(),
+}));
+
+describe('SignUpForm', () => {
+  beforeEach(() => {
+    (useCreateUserMutation as jest.Mock).mockReturnValue([jest.fn(), { isSuccess: false }]);
+  });
+
+  test('renders the SignUpForm component', () => {
+    render(<SignUpForm />);
+    expect(screen.getByText('Sign Up')).toBeInTheDocument();
+  });
+
+  test('disables the Sign Up button when form is not filled', () => {
```

Comment:
You can also test that if one of the fields is completed, the sign up button is still disabled